src/composables/walletConnect.ts

```diff
@@ -91,14 +92,26 @@ export function useWalletConnect() {
         },
       );
       if (permitted) {
+        const {
+          updateFeeList,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+        } = useEthFeeCalculation();
         const adapter = ProtocolAdapterFactory.getAdapter(PROTOCOLS.ethereum);
-        // TODO wrong data used here
-        const { hash } = await adapter.spend(
-          params.value as any,
-          params.to,
-          { maxFeePerGas: params.gas },
-        );
-        return hash;
+        if (adapter?.transferToken) {
+          await updateFeeList();
+          if (adapter.transferPreparedTransaction) {
+            const actionResult = await adapter.transferPreparedTransaction(
+              params,
+              {
+                maxPriorityFeePerGas: maxPriorityFeePerGas.value?.toFormat(ETH_COIN_PRECISION),
+                maxFeePerGas: maxFeePerGas.value?.toFormat(ETH_COIN_PRECISION),
+                fromAccount: params.from,
+              },
```

Comment:
Maybe this stuff could be calculated in the `transferPreparedTransaction` method?